pyomo/contrib/sensitivity_toolbox/sens.py

```diff
@@ -215,8 +282,37 @@ def sipopt(instance, paramSubList, perturbList,
         m.sens_init_constr[b.paramConst[kk]] = kk
         kk += 1    
     
-    
-    #Send the model to the ipopt_sens and collect the solution
-    results = opt.solve(m, keepfiles=keepfiles, tee=streamSoln)
-
-    return m
+    if method == 'sipopt':
+        # Send the model to the ipopt_sens and collect the solution
+        results = opt.solve(m, keepfiles=keepfiles, tee=streamSoln)
+    elif method == 'kaug':
+        m.dual = Suffix(direction=Suffix.IMPORT_EXPORT)
+        m.ipopt_zL_out = Suffix(direction=Suffix.IMPORT)
+        m.ipopt_zU_out = Suffix(direction=Suffix.IMPORT)
+        m.ipopt_zL_in = Suffix(direction=Suffix.EXPORT)
+        m.ipopt_zU_in = Suffix(direction=Suffix.EXPORT)    
+        m.dcdp = Suffix(direction=Suffix.EXPORT)  #: the constraint "order"  (integer >0)
+        m.DeltaP = Suffix(direction=Suffix.EXPORT)  #: the parameter values difference (float)
```

Comment:
You should consider checking the model to see if these attributes already exist (and if they are already `Suffix` components).  One route could be a helper function similar to [the one in GDP](https://github.com/Pyomo/pyomo/blob/08f4188bf5553a5aaf9856f9d5aa45ff76b4f675/pyomo/gdp/plugins/cuttingplane.py#L1038)

Reply:
I added a helper to check them. Also, removed unnecessary suffix `m.dual `

Reply:
FYI `dual` is necessary, as it provides the initial values for the dual variables in the nl file, which are used in the matrices for sensitivity calculation.